docker/ubuntu-14.04-cpu-all-options/Dockerfile

```diff
@@ -0,0 +1,72 @@
+FROM ubuntu:14.04
+MAINTAINER Aaron Markham <aaronmarkham@fb.com>
+
+# caffe2 install with cpu support
+
+########## REQUIRED DEPENDENCIES ################
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    python-dev \
+    python-pip \
+    git \
+    build-essential \
+    cmake \
+    libprotobuf-dev \
+    protobuf-compiler \
+    libgoogle-glog-dev
```

Comment:
> clean up the apt cache by removing /var/lib/apt/lists to reduce the image size, since the apt cache is not stored in a layer
https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#run